src/pytest_memray/cpython_caches.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import annotations
+
+import contextlib
+import sys
+import warnings
+from inspect import isabstract
+from typing import Any
+from typing import Generator
+
+try:
+    from _abc import _get_dump
+except ImportError:
+    import weakref
+
+    def _get_dump(cls: Any) -> tuple[set[Any], Any, Any, Any]:
+        # Reimplement _get_dump() for pure-Python implementation of
+        # the abc module (Lib/_py_abc.py)
+        registry_weakrefs = {weakref.ref(obj) for obj in cls._abc_registry}
+        return (
+            registry_weakrefs,
+            cls._abc_cache,
+            cls._abc_negative_cache,
+            cls._abc_negative_cache_version,
+        )
+
+
+@contextlib.contextmanager
+def warmup_bytecode_cache() -> Generator[None, None, None]:
+    prev_prov = sys.getprofile()
+    sys.setprofile(lambda *args: args)
```

Comment:
We should just return `None`, since the return value of a profile function is unused.